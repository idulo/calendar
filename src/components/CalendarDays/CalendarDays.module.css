.container {
  display: flex;
  justify-content: space-around;
  padding: 0.5rem;
}

.calendarDays {
  table-layout: fixed;
  width: 100%;
}

.calendarDays > td {
  width: 33.33%;
  position: relative;
}
