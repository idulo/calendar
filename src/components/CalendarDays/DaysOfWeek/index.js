import DaysOfWeek from "./DaysOfWeek";

export default DaysOfWeek;
