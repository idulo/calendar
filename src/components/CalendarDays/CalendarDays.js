import React from "react";
import moment from "moment";

import DaysOfWeek from "./DaysOfWeek";
import CalendarTable from "./CalendarTable";

import STYLES from "./CalendarDays.module.css";

class CalendarDays extends React.Component {
  firstDayOfMonth = momentDate =>
    moment(momentDate)
      .startOf("month")
      .weekday();

  daysInMonth = momentDate => moment(momentDate).daysInMonth();

  render() {
    const date = this.props.date;

    return (
      <div className={STYLES.container}>
        <table cellPadding="2" cellSpacing="2" className={STYLES.calendarDays}>
          <thead>
            <DaysOfWeek />
          </thead>
          <tbody>
            <CalendarTable
              daysInMonth={this.daysInMonth(date)}
              firstDayOfMonth={this.firstDayOfMonth(date)}
              toggleReminder={this.props.toggleReminder}
            />
          </tbody>
        </table>
      </div>
    );
  }
}

export default CalendarDays;
