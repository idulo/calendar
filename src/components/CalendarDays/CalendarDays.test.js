import React from "react";
import ReactDOM from "react-dom";
import { create } from "react-test-renderer";
import CalendarDays from "./CalendarDays";

describe("CalendarDays", () => {
  let daysInMonth = jest.fn().mockImplementation(value => 31);
  let firstDayOfMonth = jest.fn().mockImplementation(value => 2);

  it("renders without crashing", () => {
    const div = document.createElement("div");

    ReactDOM.render(
      <CalendarDays
        daysInMonth={daysInMonth}
        firstDayOfMonth={firstDayOfMonth}
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders a calendar for 31 days starting on Tuesday", () => {
    const Days = create(
      <CalendarDays
        daysInMonth={daysInMonth}
        firstDayOfMonth={firstDayOfMonth}
      />
    );

    expect(Days.toJSON()).toMatchSnapshot();
  });
});
