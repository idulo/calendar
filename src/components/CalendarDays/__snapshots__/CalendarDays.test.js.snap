// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDays renders a calendar for 31 days starting on Tuesday 1`] = `
<div
  className="container"
>
  <table
    cellPadding="2"
    cellSpacing="2"
    className="calendarDays"
  >
    <thead>
      <tr
        className="container"
      >
        <td>
          Sun
        </td>
        <td>
          Mon
        </td>
        <td>
          Tue
        </td>
        <td>
          Wed
        </td>
        <td>
          Thu
        </td>
        <td>
          Fri
        </td>
        <td>
          Sat
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          1
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          2
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          3
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          4
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          5
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          6
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          7
        </td>
      </tr>
      <tr>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          8
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          9
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          10
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          11
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          12
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          13
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          14
        </td>
      </tr>
      <tr>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          15
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          16
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          17
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          18
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          19
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          20
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          21
        </td>
      </tr>
      <tr>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          22
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          23
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          24
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          25
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          26
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          27
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          28
        </td>
      </tr>
      <tr>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          29
        </td>
        <td
          className="dayCell"
          onClick={[Function]}
        >
          30
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
